packages/ui/src/containers/SocialSignIn/SecondarySocialSignIn.test.tsx

```diff
@@ -1,12 +1,40 @@
-import { render } from '@testing-library/react';
+import { render, fireEvent, waitFor } from '@testing-library/react';
 import React from 'react';
-import { MemoryRouter } from 'react-router-dom';
+import { MemoryRouter, Route, Routes } from 'react-router-dom';
 
+import ContextProvider from '@/__mocks__/ContextProvider';
 import { socialConnectors } from '@/__mocks__/logto';
+import * as socialSignInApi from '@/apis/social';
+import { generateState, storeState } from '@/hooks/use-social';
 
 import SecondarySocialSignIn from './SecondarySocialSignIn';
 
 describe('SecondarySocialSignIn', () => {
+  const mockOrigin = 'https://logto.dev';
+
+  const invokeSocialSignInSpy = jest
+    .spyOn(socialSignInApi, 'invokeSocialSignIn')
+    .mockResolvedValue({ redirectTo: `${mockOrigin}/callback` });
+
+  const signInWithSocialSpy = jest
+    .spyOn(socialSignInApi, 'signInWithSocial')
+    .mockResolvedValue({ redirectTo: `${mockOrigin}/callback` });
+
+  beforeEach(() => {
+    /* eslint-disable @silverhand/fp/no-mutation */
+    // @ts-expect-error mock global object
+    globalThis.logtoNativeSdk = {
+      platform: 'web',
+      getPostMessage: jest.fn(() => jest.fn()),
```

Comment:
is this on purpose?

Reply:
`jest.fn(() => jest.fn())?`  Yes, its function returns a function